plugins/woocommerce-admin/client/products/fields/attribute-control/attribute-control.tsx

```diff
@@ -74,79 +94,89 @@ export const AttributeControl: React.FC< AttributeControlProps > = ( {
 		);
 	};
 
-	const onRemove = ( attribute: ProductAttribute ) => {
+	const handleRemove = ( attribute: ProductAttribute ) => {
 		// eslint-disable-next-line no-alert
 		if ( window.confirm( __( 'Remove this attribute?', 'woocommerce' ) ) ) {
-			recordEvent(
-				'product_remove_attribute_confirmation_confirm_click'
-			);
 			handleChange(
 				value.filter(
 					( attr ) =>
-						fetchAttributeId( attr ) !==
-						fetchAttributeId( attribute )
+						getAttributeId( attr ) !== getAttributeId( attribute )
 				)
 			);
-		} else {
-			recordEvent( 'product_remove_attribute_confirmation_cancel_click' );
+			onRemove( attribute );
+			return;
 		}
+		onRemoveCancel( attribute );
+	};
+
+	const openNewModal = () => {
+		setIsNewModalVisible( true );
+		onNewModalOpen();
+	};
+
+	const closeNewModal = () => {
+		setIsNewModalVisible( false );
+		onNewModalClose();
```

Comment:
I think we should check that this is a function before trying to call it (see comment above).